internal/app/admin/http/handler_test.go

```diff
@@ -0,0 +1,178 @@
+package handler
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	gcp "github.com/envoyproxy/go-control-plane/pkg/cache/v2"
+	mock_mapper "github.com/envoyproxy/xds-relay/internal/app/mapper/mock"
+	mock_orchestrator "github.com/envoyproxy/xds-relay/internal/app/orchestrator/mock"
+	bootstrapv1 "github.com/envoyproxy/xds-relay/pkg/api/bootstrap/v1"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAdminServer_DefaultHandler(t *testing.T) {
+	req, err := http.NewRequest("GET", "/", nil)
+	assert.NoError(t, err)
+
+	rr := httptest.NewRecorder()
+	handler := defaultHandler([]Handler{{
+		"/foo",
+		"does nothing",
+		http.HandlerFunc(nil),
+	}})
+
+	handler.ServeHTTP(rr, req)
+	assert.Equal(t, http.StatusOK, rr.Code)
+	assert.Equal(t, "admin commands are:\n  /foo: does nothing\n", rr.Body.String())
+}
+
+func TestAdminServer_DefaultHandler_NotFound(t *testing.T) {
+	req, err := http.NewRequest("GET", "/not-implemented", nil)
+	assert.NoError(t, err)
+
+	rr := httptest.NewRecorder()
+	handler := defaultHandler([]Handler{{
+		"/foo",
+		"does nothing",
+		http.HandlerFunc(nil),
+	}})
+
+	handler.ServeHTTP(rr, req)
+	assert.Equal(t, http.StatusNotFound, rr.Code)
+	assert.Equal(t, "404 page not found\n", rr.Body.String())
+}
+
+func TestAdminServer_ConfigDumpHandler(t *testing.T) {
+	req, err := http.NewRequest("GET", "/server_info", nil)
+	assert.NoError(t, err)
+
+	rr := httptest.NewRecorder()
+	handler := configDumpHandler(&bootstrapv1.Bootstrap{
+		Server: &bootstrapv1.Server{Address: &bootstrapv1.SocketAddress{
+			Address:   "127.0.0.1",
+			PortValue: 9991,
+		}},
+		OriginServer: nil,
+		Logging:      nil,
+		Cache:        nil,
+	})
+
+	handler.ServeHTTP(rr, req)
+	assert.Equal(t, http.StatusOK, rr.Code)
+	assert.Equal(t,
+		`{
+  "server": {
+    "address": {
+      "address": "127.0.0.1",
```

Comment:
is there any chance the order of the the `server.address` fields are swapped? If so, how about we compare the deserialized json instead?

Reply:
can you clarify what you mean by swapped?

Reply:
Sure. I was asking about the possibility of `port_value` coming before `address`. In case that's possible, that strict comparison would break, right?

Reply:
I think it gets unmarshalled based on the order of the fields in the struct, as suggested by https://blog.golang.org/json.